### ISSUE TITLE
Repository has binaries

### ISSUE THREAD

**Issue by gdt** [sparkfun/SparkFun_RTK_Firmware#155]
I realize this code is hard for a lot of people to build, so I don't mean to suggest in the slightest that you not distribute binaries.

I would expect this to be a repository of source code, and for builds of the source to be available some other way, download dir, build artifacts from GHA integration, or something else.  The github way would be to have a GHA set up that installs the toolchain and builds and packages artifacts, so that each commit has binaries, but that they aren't in the repo.  This would also remove a lot of manual work.

Besides an awkwardness of binaries in a source place:
  - when a release is tagged, presumably it is then built from the tag and then committed, so the tag won't include the binary
  - binaries don't compress well against each other.  The repo already takes 0.5G and with the (really great to see and much appreciated!) rate of progress, it seems likely to get bigger fairly fast.

**Comment by nseidle** [comment]
*Thank you* for the feedback. If it wasn't apparent, we are github apprentices attempting to be journeymen someday.

> GHA Integration

I have been thinking about this as well. It would be really nice to have weekly or nightly builds. How do you propose we do this? I have some [limited experience](https://github.com/sparkfun/SparkFun_VL53L1X_Arduino_Library/blob/master/.github/workflows/LibraryBuild.yml) with yaml workflow building. Can we build a yml that also pulls in Arduino libraries that are not true libraries (currently there are [two libs](https://sparkfun.github.io/SparkFun_RTK_Firmware/firmware_update/#compiling-from-source) that are on github but not technically a library)

I welcome any changes you recommend to the repo to make it more compressed. I don't predict commit rates will slow down anytime soon.

**Comment by gdt** [comment]
I am not an expert at GHA but in unison, I end up merging work by others which involves installing ocaml and other things.  Basically I think you can do anything you can do in a shell including downloading the libraries.    Of course it needs to be command line, not a GUI.    I would recommend doing this with ubuntu as in unison we have found that the CI runs in Windows are flaky.

The normal setup runs the CI step after every commit, so there's no need for weekly.   And if the runs mark artifacts they become available in a release tag.   With that, the compiled binaries could be removed.

**Comment by nseidle** [comment]
Looking through the 100MB zip of the repo, a big chunk is the recently added QtPy GUI executables (~30MB each). But I agree, if we keep adding a 2MB binary every few weeks it will get out of hand.

**Comment by tonycanike** [comment]
Maybe a little step can be made in the right direction by not adding anymore binaries to the repo starting today?   Instead host the binaries, old and new, somewhere else (sparkfun.com?) as an intermediate step. This would significantly slow down the repo growth rate.

Put a link in the repo to wherever the binaries are hosted would let everyone find them.

Weekly or nightly builds would be very helpful to folks (like me) who'd be happy to test the builds. I haven't gotten around to setting up my own build environment and may never get to it.

**Comment by nseidle** [comment]
Good news:

* We've determined we *don't* need separate binaries for the 4MB devices. The GUI handles selecting the partition file correctly for 16MB or 4MB devices.
* We've got Github actions and Arduino CLI working together to compile our firmware. I haven't turned on nightly builds yet (just because I leave the RC in a broken state too often) but it's possible someday.

**Comment by gdt** [comment]
Based on experience with another project, I would suggest
  - don't configure nightly builds
  - run build/test (whatever is done) for any commit on release_candidate (aka "develop" :-) and master
  - run build/test on pull requests
  - don't merge PRs unless CI passes
  - move to this new world as soon as possible; it will make a big difference i think

**Comment by nseidle** [comment]
Binaries are now located on a [separate repo](https://github.com/sparkfun/SparkFun_RTK_Firmware_Binaries) and on a given [release](https://github.com/sparkfun/SparkFun_RTK_Firmware/releases). I moved the python tools over as well as I suspect most folks will be grabbing the windows executables directly without the need to download the entire repo. 

That leaves this repo at 25MB. Not great, and I worry that the docs images will continue to creep up, but it's much better than >100MB. Thanks for the poke @gdt.
